wordpress-mobile/WordPress-iOS
# Implement featured image with a relationship between Post and Media objects.

This PR implements the setting of the featured image on post in a way that is compatible with the new asyc publish os posts.

It works by making the `featuredImage` attribute on `AbstractPost` a full CoreData relationship instead of piggy backing on the post_thumbnail attribute.

This allows to set the featuredImage even before the upload of Media is started and the allows to publish the post without waiting for that upload to finish.

To test:
 - Start the app
 - Create a new post
 - Open post options
 - Set a featured image on it
 - Before the featured image is set
 - Tap Back
 - Tap Publish
 - See the Post being published 
 - Go back to the post and see if the featured image is set properly
 - Try the same but without leaving the editor interface while the upload is happening
 - Open the post list on a blog that you never opened before on the app or from a fresh install
 - Open a post that you know has a featured image set
 - Check the post settings to see if the featured image is loaded correctly
